snapcraft/internal/lxd.py

```diff
@@ -165,18 +165,25 @@ def _inject_snap(self, name):
         except Exception as e:
             raise SnapcraftEnvironmentError(
                 'Error connecting to {}'.format(api)) from e
-        if not json['status'] == 'OK':
+        if json['status'] == 'error':
             raise SnapcraftEnvironmentError(
-                'Error querying {r!} snap: {}'.format(name, json))
+                'Error querying {!r} snap: {}'.format(
+                    name, json['result']['message']))
         classic = json['result']['confinement'] == 'classic'
         rev = json['result']['revision']
 
+        # https://github.com/snapcore/snapd/blob/master/snap/info.go
+        # MountFile
+        filename = '{}_{}.snap'.format(name, rev)
+        # https://github.com/snapcore/snapd/blob/master/dirs/dirs.go
+        # CoreLibExecDir
         installed = os.path.join(os.path.sep, 'var', 'lib', 'snapd', 'snaps',
-                                 '{}_{}.snap'.format(name, rev))
-        filename = os.path.basename(installed)
-        # Copy file to ensure LXD snap can access it
-        copyfile(installed, filename)
-        self._push_file(filename, os.path.join(self._project_folder, filename))
+                                 filename)
+        # Copy file to a path the 'lxd' snap can access
+        filepath = os.path.join(os.path.sep, 'run', 'user', str(os.getuid()),
+                                'snap.lxd', filename)
+        copyfile(installed, filepath)
```

Comment:
What if `dirname(filepath)` doesn't exist (e.g. lxd is not running from the snap)? At least from the [docs](https://docs.python.org/3/library/shutil.html#shutil.copyfile) it seems copyfile will not create it.